Source/Core/Resource.js

```diff
@@ -1869,24 +1868,26 @@ define([
 
                 return Resource.fetchBlob({
                     url: url
-                });
-            })
-            .then(function(blob) {
-                if (!defined(blob)) {
-                    return;
-                }
+                })
+                .then(function(blob) {
+                    if (!defined(blob)) {
+                        deferred.reject('Failed to fetch blob.');
```

Comment:
As a general rule, you should always throw/reject an `Error` object, never a string or something else.  In this case, it probably makes sense to use `new RuntimeError()` with your message.

Instead of `Failed to fetch blob` which sounds like a networking issue consider something  more direct and helpful to the user like "Successfully retrieved {url} but it contained no content."